cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1018,27 +1016,28 @@ if(builtin_xrootd)
   set(xrootd ON CACHE BOOL "Enabled because builtin_xrootd requested (${xrootd_description})" FORCE)
 endif()
 
-if(xrootd AND XROOTD_VERSIONNUM VERSION_GREATER_EQUAL 500000000)
-  if(xproofd)
-    if(fail-on-missing)
-      message(FATAL_ERROR "XROOTD is version 5 or greater. The legacy xproofd servers can not be built with this version. Use -Dxproofd:BOOL=OFF to disable.")
-    else()
-      message(STATUS "XROOTD is version 5 or greater. The legacy xproofd servers can not be built with this version. Disabling 'xproofd' option.")
-      set(xproofd OFF CACHE BOOL "Disabled because xrootd version is 5 or greater" FORCE)
-    endif()
-  endif()
-endif()
-
 #---check if netxng and netx can be built-------------------------------
-if(xrootd AND XROOTD_VERSIONNUM VERSION_GREATER 300030005)
+if(xrootd)
+  # Requires XRootD 3.3.5, which should be a given at this point.
   set(netxng ON)
-else()
-  set(netxng OFF)
-endif()
-if(xrootd AND XROOTD_VERSIONNUM VERSION_LESS 500000000)
-  set(netx ON)
-else()
-  set(netx OFF)
+
+  # By the presence of the XrdClient/XrdClientAdmin.hh header, we check if netx can be compiled (XRootD < 5.0.0).
+  set(CMAKE_REQUIRED_INCLUDES ${XROOTD_INCLUDE_DIRS})
+  check_include_file_cxx(XrdClient/XrdClientAdmin.hh _xrd_client_admin_hh)
```

Comment:
This just accidentally might find header files in system locations? Can you add a if (xrootd VERSION_LESS 5) around this or something?

cf. #11750 where libraries from different xrootd installations were mixed.

Reply:
Thank you, that's a great idea!

Reply:
Ah, now I remember why I didn't do this obvious thing! It's because XRootD only includes the version info in the latest releases:
https://github.com/xrootd/xrootd/pull/2015

Until then, we either keep a more rigurous version check line in our old `FindXROOTD.cmake`, or we use a less verbose hack like this. Considering that XRootD is already at its [end of support](https://github.com/root-project/root/issues/12631#issuecomment-1740452464), I don't think it's worth to spend much time optimizing this. I think in ROOT we are anyway going to remove soon the components that require XRootD 4.

But maybe @amadio has a better idea!

Reply:
But the problem really is that you might have more than one xrootd in your paths somehow, so you really shouldn't pick up these header files if you already know that you are not compatible.

Something like?
```cmake
# if we don't have any version information from the XROOTD cmake, let's just try 
# to pick up some header files
if(NOT XRootD_VERSION_MAJOR or XRootD_VERSION_MAJOR VERSION_LESS 5)
#...
endif()
```

Reply:
I see, yes that makes sense. So in that case, you can be sure that at least if you have a new xrootd with version variables (which I presume you do in LCG, right?), then the old headers will not be detected.

Reply:
Yes, we have a fairly recent xrootd (5.6.0)

Reply:
Nice, and thanks again for your feedback! I think I addressed your issue now, can you check if #11750 is fixed?

Now, if you have `xrootd > 5.6.0` installed and you don't use `builtin_xrootd`, there should be nothing in the cmake code path besides standard `find_package()` and version checks.

Reply:
I think for now it might be necessary to keep your own module, but simply require a new version and drop all deprecated code using XRootD 4.x. I will fix `XRootDConfig.cmake` to properly export a version in the next feature release, then you will be able to call `find_package(XRootD 5.7)` and finally drop `FindXROOTD.cmake` from ROOT.

Reply:
I've created a ticket on XRootD for this: xrootd/xrootd#2094

Reply:
That's great, thank you!